src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -629,8 +629,31 @@ private bool StartWithCreateProcess(ProcessStartInfo startInfo)
             return ret;
         }
 
-        private static Encoding GetEncoding(int codePage)
+        private static Encoding GetConsoleEncodingOrThrow(ConsoleStreamDirection encodingType)
         {
+            const int INVALID_HANDLE = 6;
+            int codePage;
+            try
+            {
+                codePage = (encodingType == ConsoleStreamDirection.Input) ?
+                    (int)Interop.Kernel32.GetConsoleCP() :
+                    (int)Interop.Kernel32.GetConsoleOutputCP();
+
+                int error = Marshal.GetLastWin32Error();
+                if (error == INVALID_HANDLE)
```

Comment:
We update the assembly version if we need to support this in the future. I don't like the runtime check because if this failed on non UAP and throwing such exception would not be nice.

do we know why this is not supported in the UAP today? should we ask Windows to support it?